schaetzmeisterin-backend-server/src/main/java/de/fabiankeck/schaetzmeisterinbackendserver/Service/GameService.java

```diff
@@ -0,0 +1,53 @@
+package de.fabiankeck.schaetzmeisterinbackendserver.Service;
+
+import de.fabiankeck.schaetzmeisterinbackendserver.dto.SignInUserDto;
+import de.fabiankeck.schaetzmeisterinbackendserver.model.Game;
+import de.fabiankeck.schaetzmeisterinbackendserver.model.Player;
+import de.fabiankeck.schaetzmeisterinbackendserver.utils.IdUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+
+@Service
+public class GameService {
+    private final List<Game> games = new ArrayList<>();
+    private final IdUtils idUtils;
+
+    public GameService(IdUtils idUtils) {
+        this.idUtils = idUtils;
+    }
+
+
+    public Game initNewGame() {
+        Game newGame = new Game(idUtils.createId(), List.of());
+        games.add(newGame);
+        return newGame;
+    }
+
+    public Game userSignIn(Optional<String> gameId, SignInUserDto signInUserDto) {
+        Game game;
+        if(gameId.isEmpty()){
+            game = initNewGame();
+        } else {
+            game = findByID(gameId.get()).orElseThrow(()-> new ResponseStatusException(HttpStatus.NOT_FOUND));
+        }
```

Comment:
i would extract this to an own method and return the game. instead of reassign

Reply:
more like in the latest commit?